lib/components/hikingmapbox.dart

```diff
@@ -167,30 +168,66 @@ class _MapWidgetState extends State<MapWidget> {
   }
 
   void updateRoute() async {
-    List<LatLng> remainingRoute = [];
-    List<LatLng> passedRoute = _passedRoute;
-
-    for (int i = 0; i < _route.length; i++) {
-      LatLng latLng = _route[i];
-      double distanceToCurrentLocation =
-          OsmData.getDistance(latLng, _currentDeviceLocation);
-      if (distanceToCurrentLocation > 0.0002) {
-        remainingRoute.add(latLng);
-      } else {
-        passedRoute.add(latLng);
+    int deleteAllRouteNodesWithAnIndexSmallerThan = 0;
```

Comment:
personal preference:
in my opinion this is a really weird variable name, it is kind of descriptive, but variables don't delete anything. Therefore i would suggest to make it simpler and shorter:
You are trying to find the index at which way point the user is currently located why not something like:
**currentRouteNodeIndex** or even **currentNodeIndex**? In this scenario it is clear that the node index refers to the route.